### PR TITLE
Add a --fast-deps option to `comfy install/reinstall` that enables new faster/unified/uv-based dependency install

### PR DESCRIPTION
**Pull request by telamonian**
This PR adds a new option `--fast-deps`, to `comfy install` and `comfy reinstall`. This option enables the new dependency handling implementation. The features of said new dependency handling currently cover:

- Uses uv to install all Python deps (very fast)
- Resolves all core and custom extension dependencies together
  - Any conflicts between core and extensions are settled in core's favor
  - TODO: extension-vs-extension dependency conflict handling
- Minimizes total number of subprocess calls to external Python dependency-handling tools

This PR depends on ltdrdata/ComfyUI-Manager#886, which adds support for a `--no-deps` option to the lower level cm-cli interface. Although ComfyUI-Manager#886 has been merged, the relevant code has not yet been pushed to the main branch of ComfyUI-Manager. So if you want to test this PR out yourself, you'll have to manually switch to the `feat/cnr` branch on the instance of ComfyUI-Manager that comfy-cli installs.

### CODE REVIEW

**Comment by telamonian** [comment]
Okay, the code in this PR is coming along. Fixed a bunch of bugs/problems, and thanks to #147 this PR is now much easier to test out for yourself. Just install this PR branch, then do:

```bash
comfy install --fast-deps --manager-url=https://github.com/ltdrdata/ComfyUI-Manager.git@feat/cnr
```

specifying any other args as normal. The fast-deps option will now also work with installing custom nodes, eg:

```bash
comfy node install --fast-deps comfyui-kjnodes comfyui-animatediff-evolved
```

**Comment by telamonian** [comment]
Here's the features/TODO list for the PR:

- [x] add `--fast-deps` option to trigger the new install behavior
  - [ ] (optional) ensure that all possible relevant commands support `--fast-deps`
    - [x] `[node] install`
    - [x] `[node] reinstall`
    - [ ] others?  
- [x] speed up python dependency installs
  - [x] write uv-based installer
  - [x] minimize calls to subprocess/batch all possible installs
- [ ] Compile all dependencies to explicit versions before install
  - [x] pre-compile step to calculate overrides
  - [x] unified compile (core + exts)
- [ ] UI/UX for conflicts
  - [x] core-vs-extension conflicts are always settled in favor of core
  - [ ] resolve extension-vs-extension conflicts via at least one of:
    - [ ] stop, notify user of which extensions conflict (currently not possible due to upstream issue, see https://github.com/astral-sh/uv/issues/1854)
    - [ ] ask for user input, show selector with possible versions, default to latest
- [x] add special case handling 
- [ ] finalize tests

So the big remaining issues are adding some kind of extension-vs-extension conflict resolution and finishing up any tests for the new installer. Unfortunately the obvious easy path to implement ext-vs-ext behavior is a no-go; due to an upstream issue there's no way currently for us to report to the user which extensions are actually in conflict, so we can't just print out the relevant info and then tell the user to manually resolve the conflict

**Review by robinjhuang — CHANGES_REQUESTED** [review]

**Comment by telamonian** [comment]
Okay, unittest is now functional. Said test needs a small fix tho (I realized that currently it will break whenver sympy or numpy have a new release, which is not ideal). I'll add the fix and address all of the feedback tomorrow, and then I'll merge

**Comment by robinjhuang** [comment]
> Okay, unittest is now functional. Said test needs a small fix tho (I realized that currently it will break whenver sympy or numpy have a new release, which is not ideal). I'll add the fix and address all of the feedback tomorrow, and then I'll merge

Sounds great!

File: comfy_cli/command/install.py [hunk 1]
```diff
@@ -208,9 +209,10 @@ def execute(
         os.chdir(repo_dir)
         subprocess.run(["git", "checkout", commit], check=True)
 
-    install_comfyui_dependencies(
-        repo_dir, gpu, plat, cuda_version, skip_torch_or_directml, skip_requirement
-    )
+    if not fast_deps:
+        install_comfyui_dependencies(
```
[the remark this answers is not in the document]
Comment (telamonian) [reply comfy_cli/command/install.py]: done!

File: comfy_cli/command/install.py [hunk 2]
```diff
@@ -236,13 +238,18 @@ def execute(
...
                 # clone specific branch
                 manager_url, manager_branch = manager_url.rsplit("@", 1)
-
                 subprocess.run(["git", "clone", "-b", manager_branch, manager_url, manager_repo_dir], check=True)
             else:
                 subprocess.run(["git", "clone", manager_url, manager_repo_dir], check=True)
 
-            install_manager_dependencies(repo_dir)
+            if not fast_deps:
+                install_manager_dependencies(repo_dir)
```
[the remark this answers is not in the document]
Comment (telamonian) [reply comfy_cli/command/install.py]: and done!

File: comfy_cli/uv.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+        self.gpu = DependencyCompiler.ResolveGpu(gpu)
+
+        self.gpuUrl = DependencyCompiler.nvidiaPytorchUrl if self.gpu == GPU_OPTION.NVIDIA else DependencyCompiler.rocmPytorchUrl if self.gpu == GPU_OPTION.AMD else None
+        self.out = self.cwd / outName
+        self.override = self.cwd / "override.txt"
+
+        self.reqFilesCore = reqFilesCore if reqFilesCore is not None else self.findCoreReqs()
+        self.reqFilesExt = reqFilesExt if reqFilesExt is not None else self.findExtReqs()
+
+    def findCoreReqs(self):
```
[the remark this answers is not in the document]
Comment (telamonian) [reply comfy_cli/uv.py]: done!

**Pull request merged**